spec/spec_helper.rb

```diff
@@ -3,6 +3,8 @@
 require "coveralls"
 Coveralls.wear!("rails")
 
+require "chipmunk"
```

Comment:
Requiring this here breaks test isolation, and obscures any issues with missing requires in the non-test code.

Reply:
Hmmm... But requires are for the lifetime of the suite, and so, spec files are not genuinely isolated to begin with.

The theory here is that we should treat the "chipmunk library" like a gem. We `require "chipmunk"` and get the whole library. Granted, there are gems that have multiple modules or optional pieces, but I don't think there will be enough code (or downstream dependencies like drivers) here to warrant the performance/confusion tradeoff of partial loading now.

We `require "chipmunk"` in the services initializer in the app. That will be run in anything that would "boot" the app (including rake tasks with an 'environment' dependency). I think requiring it here is _more_ consistent and deterministic in the big picture than scattering loads across VM uptime.

To compare with Java, this "require it now" approach is more like adding a jar to the classpath. The imports for name resolution are entirely unrelated to loading. Doing incremental requires during runtime is more like `Class.forName`, which can be powerful for dynamic dependencies, but can also cause nondeterminism if not managed carefully.